algorithm/main.go

```diff
@@ -0,0 +1,62 @@
+// Package algorithm provides functionalities to interact crystal
+package algorithm
+
+import (
+	"fmt"
+
+	"github.com/SneaksAndData/esd-services-api-client-go/shared/http"
+)
+
+// Service encapsulates the HTTP client and URLs needed to interact with the algorithm service.
+type Service struct {
+	httpClient   *http.Client
+	schedulerURL string
+	receiverURL  string
+	apiVersion   string
+}
+
+// Payload defines the structure of the request body for creating algorithm runs.
+type Payload struct {
+	AlgorithmParameters interface{}
```

Comment:
can't this one be defined as `map[string]string`? This is what it is essentially

Reply:
map[string]string will not work in this case since we can have properties that are lists. I've changed it to  map[string]interface{}